packages/download's ensureManagedBase() has the same name-only OS-artifact check as pre-fix apps/desktop (no lstat type verification)

## Summary

PR #7267 fixed a defect in `apps/desktop/src/main/updater/store.ts` where `isOsManagedRootArtifact()` matched `.DS_Store`/`Thumbs.db`/`desktop.ini`/`.localized` by name only, with no `lstat` check confirming the entry is actually a plain file. That let a directory or symlink squatting on one of those names be silently treated as harmless OS litter — bypassing the empty-root claim check and staying permanently excluded from unexpected-entry corruption validation.

`packages/download/src/store.ts`'s `ensureManagedBase()` has the identical unpatched pattern. It was already flagged once as a follow-up (#7266 → merged via #7268), but that merge duplicated the same name-only check rather than the type-verified version:

```ts
// packages/download/src/store.ts:27-34 (current upstream/main)
const OS_MANAGED_ROOT_ARTIFACTS = new Set([".DS_Store", "Thumbs.db", "desktop.ini", ".localized"]);
function isOsManagedRootArtifact(name: string): boolean {
  return OS_MANAGED_ROOT_ARTIFACTS.has(name);
}
// line 107, inside ensureManagedBase():
const content = entries.filter((name) => !isOsManagedRootArtifact(name));
```

A fresh managed-download base whose only entry is a directory (or symlink) named `.DS_Store` would be silently claimed as empty and owned.

## Proposed direction

Mirror #7267's fix (`apps/desktop/src/main/updater/store.ts`'s `verifiedOsManagedRootArtifacts()`): before trusting a name match, `lstat` the entry and only allow it through when it's a plain file, not a directory or symlink. `packages/download` can't import `apps/desktop`'s helper directly (dependency direction is the other way), so this needs its own local implementation, same as the existing duplicated constant/predicate.

Add regression tests mirroring the two added in #7267:
- a directory named `.DS_Store` at a fresh base must not be claimed as owned
- a symlink named `.DS_Store` at an already-owned base must still be reported as content, not silently ignored

## Related, minor cleanup (same review pass)

`apps/desktop/src/main/updater/support.ts`'s `directoryIsEmpty()` became dead code as a side effect of #7267's diff (its only caller in `store.ts` was replaced). It's a trivial removal, but is worth doing here since it's the exact same name-only pattern (`entries.length === 0`, no OS-artifact or type awareness) sitting unused right next to the fix — leaving it around risks someone reaching for it again later.

## Context

- Found during independent adversarial re-review while following up on PR #7267.
- Not blocking #7267, which only touches `apps/desktop`.

Implemented and opened PR #7603: https://github.com/nexu-io/open-design/pull/7603

The download store now verifies that an OS-managed root artifact is a regular file before ignoring it. OS-artifact directories and symlinks are therefore treated as unexpected content in a fresh managed base, and regression coverage was added in `packages/download/tests/index.test.ts`.

Validation:
- `git diff --check` passed.
- The focused test could not be run locally because the workspace install was blocked by repeated pnpm `ERR_PNPM_EPERM` import failures, and the checkout reports Node.js 22 while the repository requires Node.js 24.
- Remote CI passed the relevant preflight, workspace unit tests, and Playwright critical (b) jobs.
- Remote CI failed in the unrelated Playwright critical (a) `deep-link-preview` test, which timed out waiting for the existing `Loading OpenDesign…` screen to clear after 24 other tests passed. The failure is in `e2e/ui/app.test.ts` and does not exercise the download package change.

The implementation is contained in commit `50578c6075f4c11d612c8990ac97647298e017df`.

--- a/packages/download/src/store.ts
+++ b/packages/download/src/store.ts
@@ -23,6 +23,11 @@ import { readJson, writeJson } from "./fs-io.js";
  * Thumbs.db/desktop.ini. Mirrors apps/desktop's isOsManagedRootArtifact —
  * duplicated rather than shared since apps/desktop depends on this package,
  * not the reverse.
+ *
+ * Matched by name only — callers MUST additionally confirm the entry is a
+ * plain file via {@link verifiedOsManagedRootArtifacts} before trusting a
+ * match. A directory or symlink squatting on one of these names must still
+ * be treated as real content, not waved through as harmless OS litter.
  */
 const OS_MANAGED_ROOT_ARTIFACTS = new Set([".DS_Store", "Thumbs.db", "desktop.ini", ".localized"]);
 
@@ -32,6 +37,31 @@ const OS_MANAGED_ROOT_ARTIFACTS = new Set([".DS_Store", "Thumbs.db", "desktop.in
  */
 function isOsManagedRootArtifact(name: string): boolean {
   return OS_MANAGED_ROOT_ARTIFACTS.has(name);
+}
+
+/**
+ * @internal Resolves which of `entries` (direct children of `basePath`) are
+ * genuine OS-managed artifacts: name-matched AND a plain file, not a
+ * directory or symlink. A name match alone is not enough — see {@link
+ * isOsManagedRootArtifact}. An entry that disappears or fails to stat
+ * between `readdir` and this check is treated as unverified (excluded),
+ * which fails closed into the existing rejection path rather than silently
+ * trusting it. Mirrors apps/desktop's verifiedOsManagedRootArtifacts.
+ */
+async function verifiedOsManagedRootArtifacts(basePath: string, entries: string[]): Promise<Set<string>> {
+  const candidates = entries.filter((entry) => isOsManagedRootArtifact(entry));
+  if (candidates.length === 0) return new Set();
+  const verified = await Promise.all(
+    candidates.map(async (entry) => {
+      try {
+        const entryStat = await lstat(join(basePath, entry));
+        return entryStat.isFile() && !entryStat.isSymbolicLink() ? entry : null;
+      } catch {
+        return null;
+      }
+    }),
+  );
+  return new Set(verified.filter((entry): entry is string => entry != null));
 }
 
 /**
@@ -104,7 +134,8 @@ export async function ensureManagedBase(basePath: string): Promise<void> {
   const sentinel = await readJson<unknown>(sentinelPath);
   if (sentinel == null) {
     const entries = await readdir(basePath);
-    const content = entries.filter((name) => !isOsManagedRootArtifact(name));
+    const osArtifacts = await verifiedOsManagedRootArtifacts(basePath, entries);
+    const content = entries.filter((name) => !osArtifacts.has(name));
     if (content.length > 0) {
       throw new ManagedDownloadError(MANAGED_DOWNLOAD_ERROR_CODES.STORE_NOT_OWNED, `download base is not empty and has no ownership marker: ${basePath}`);
     }

--- a/packages/download/tests/index.test.ts
+++ b/packages/download/tests/index.test.ts
@@ -1,7 +1,7 @@
 import { createHash } from "node:crypto";
 import { spawnSync } from "node:child_process";
 import { createServer, type IncomingMessage, type ServerResponse } from "node:http";
-import { existsSync, mkdirSync, mkdtempSync, readFileSync, rmSync, writeFileSync } from "node:fs";
+import { existsSync, mkdirSync, mkdtempSync, readFileSync, rmSync, symlinkSync, writeFileSync } from "node:fs";
 import { tmpdir } from "node:os";
 import { join } from "node:path";
 import { setTimeout as sleep } from "node:timers/promises";
@@ -480,5 +480,35 @@ describe("managed download package", () => {
       code: MANAGED_DOWNLOAD_ERROR_CODES.STORE_NOT_OWNED,
     });
     expect(existsSync(join(basePath, ".open-design-download-root.json"))).toBe(false);
+  });
+
+  it("does not claim a fresh managed base whose only entry is a directory named .DS_Store", async () => {
+    const root = tmpRoot("os-artifact-dir-not-file");
+    const basePath = join(root, "downloads");
+    mkdirSync(basePath, { recursive: true });
+    mkdirSync(join(basePath, ".DS_Store"));
+    writeFileSync(join(basePath, ".DS_Store", "hidden.txt"), "not actually OS litter");
+
+    await expect(pruneManagedDownloads({ basePath })).rejects.toMatchObject({
+      code: MANAGED_DOWNLOAD_ERROR_CODES.STORE_NOT_OWNED,
+    });
+    expect(existsSync(join(basePath, ".open-design-download-root.json"))).toBe(false);
+    expect(existsSync(join(basePath, ".DS_Store", "hidden.txt"))).toBe(true);
+  });
+
+  const symlinkOsArtifactIt = process.platform === "win32" ? it.skip : it;
+  symlinkOsArtifactIt("does not extend the OS-artifact allowance to a symlink named .DS_Store", async () => {
+    const root = tmpRoot("os-artifact-symlink-not-file");
+    const basePath = join(root, "downloads");
+    const outsideTarget = join(root, "outside.txt");
+    mkdirSync(basePath, { recursive: true });
+    writeFileSync(outsideTarget, "outside content");
+    symlinkSync(outsideTarget, join(basePath, ".DS_Store"));
+
+    await expect(pruneManagedDownloads({ basePath })).rejects.toMatchObject({
+      code: MANAGED_DOWNLOAD_ERROR_CODES.STORE_NOT_OWNED,
+    });
+    expect(existsSync(join(basePath, ".open-design-download-root.json"))).toBe(false);
+    expect(existsSync(outsideTarget)).toBe(true);
   });
 });

